### ISSUE TITLE
Is this a bug? Calculating `cross_entropy loss` does not change ignore_index parameter; the default value is -100, but [PAD] token's index = 0

### ISSUE THREAD

**Issue by myhz0606** [lukas-blecher/LaTeX-OCR#305]
x-transformers version: 0.15.0

parameter meaning:  (https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html)
ignore_index ([int](https://docs.python.org/3/library/functions.html#int), optional) – Specifies a target value that is ignored and does not contribute to the input gradient. When size_average is True, the loss is averaged over non-ignored targets. Note that ignore_index is only applicable when the target contains class indices.

`class AutoregressiveWrapper(nn.Module):
    def __init__(self, net, ignore_index = -100, pad_value = 0):
        super().__init__()
        self.pad_value = pad_value
        self.ignore_index = ignore_index

        self.net = net
        self.max_seq_len = net.max_seq_len

    def forward(self, x, **kwargs):
        xi = x[:, :-1]
        xo = x[:, 1:]
        # help auto-solve a frequent area of confusion around input masks in auto-regressive
        # if user supplies a mask that is only off by one from the source sequence, resolve it for them
        mask = kwargs.get('mask', None)
        if mask is not None and mask.shape[1] == x.shape[1]:
            mask = mask[:, :-1]
            kwargs['mask'] = mask

        out = self.net(xi, **kwargs)
        loss = F.cross_entropy(out.transpose(1, 2), xo, ignore_index = self.ignore_index)
        return loss

`
